moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -1132,88 +1182,63 @@ bool MoveItConfigData::inputROSControllersYAML(const std::string& file_path)
   std::ifstream input_stream(file_path.c_str());
   if (!input_stream.good())
   {
-    ROS_ERROR_STREAM_NAMED("ros_controller.yaml", "Unable to open file for reading " << file_path);
+    ROS_WARN_STREAM_NAMED("ros_controller.yaml", "Does not exist " << file_path);
     return false;
   }
 
   // Begin parsing
   try
   {
     YAML::Node doc = YAML::Load(input_stream);
-    // Read the robot name
 
-    for (YAML::const_iterator controller_it = doc.begin(); controller_it != doc.end(); ++controller_it)
+    for (YAML::const_iterator doc_map_it = doc.begin(); doc_map_it != doc.end(); ++doc_map_it)
     {
-      if (controller_it->second.IsMap())
+      if (const YAML::Node& controllers = doc_map_it->second)
       {
+        // Used in parsing ROS controllers
+        ROSControlConfig control_setting;
+
         // Loop through all controllers
-        if (const YAML::Node& controller = controller_it->second)
+        for (YAML::const_iterator controller_it = controllers.begin(); controller_it != controllers.end();
+             ++controller_it)
```

Comment:
move the contents of this for loop into a helper function to make this more readable... named perhaps ``processROSController(controller_it, ...)``